drivers/usb/dwc2/platform.c

```diff
@@ -468,6 +471,31 @@ static int dwc2_driver_probe(struct platform_device *dev)
 	if (retval)
 		return retval;
 
+	/* Temporary workaround for USB FS clock and PHY initialization on STM32H7. */
+	np = of_find_node_by_path("/");
+	if(np)
+	{
+		if(!of_property_read_string(np, "compatible", &compat))
+		{
+			if (strstr(compat, "st,stm32h7"))
+			{
+				/* Select PLL3 as a clock source for USBOTG2. */
+				*(volatile int*)0x58024454 &= ~0x00300000;
```

Comment:
Please introduce macros for registers and bits to make the code more readable (as you did on U-Boot)